tests/order.py

```diff
@@ -210,3 +205,65 @@ def test_add_item_to_new_open_order(self):
 
         self.assertEqual(response.status_code, status.HTTP_200_OK)
         self.assertEqual(len(json_response), 2)
+
+    def test_complete_order(self):
+        """
+        Ensure we can complete an order by updating the payment.
+        """
+        # Add product to order
+        url = "/cart"
+        data = { 'product_id': self.product["id"] }
+        self.client.credentials(HTTP_AUTHORIZATION='Token ' + self.token)
+        response = self.client.post(url, data, format='json')
+
+        self.assertEqual(response.status_code, status.HTTP_201_CREATED)
+
+        # Get cart and verify product was added
+        url = "/cart"
+        self.client.credentials(HTTP_AUTHORIZATION='Token ' + self.token)
+        response = self.client.get(url, None, format='json')
+        self.order = json.loads(response.content)
+        json_response = json.loads(response.content)
+        
+
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertEqual(json_response["id"], 1)
+        self.assertEqual(json_response["size"], 1)
+        self.assertEqual(len(json_response["lineitems"]), 1)
+
+
+        # Update order with payment 
+
+        url = f"/orders/{self.order["id"]}"
+        data = { "payment_type": self.paymenttype["id"]} 
+        self.client.credentials(HTTP_AUTHORIZATION='Token ' + self.token)
+        response = self.client.put(url, data, format='json')
+        self.assertEqual(response.status_code, status.HTTP_204_NO_CONTENT)
+
+        # Get order and verify payment was added
+        url = f"/orders/{self.order["id"]}"
+        self.client.credentials(HTTP_AUTHORIZATION='Token ' + self.token)
+        response = self.client.get(url, None, format='json')
+        self.customer = json_response["complete_customer"]
```

Comment:
Did you mean to reassign self.customer here? It is originally set in the setUp function, and I think it would make more sense to use the setUp version of customer when running your assertEqual function on line 253.

Reply:
You were absolutely right I removed that line and was able to reference the self.customer from the setup. Thanks!